### PR TITLE
PXB-2258 document --lock-ddl and --lock-ddl-per-table (2.4)

### PR DESCRIPTION
**Pull request by patrickbirch**
new file:   source/advanced/locks.rst
	modified:   source/faq.rst
	modified:   source/index.rst

### CODE REVIEW

**Review by kathy-williamson — APPROVED** [review]

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+===========================================
+``lock-ddl-per-table`` Option Improvements
+===========================================
+
+|MySQL| 5.7 has made improvements to bulk loading, one of these improvements is the ability to disable redo logging, which increased the speed of ADD INDEX operations.
+
+To block DDL statements on an instance, Percona Server implemented LOCK TABLES FOR BACKUP. |Percona XtraBackup| uses this lock for the duration of the backup. This lock does not affect DML statements.
+
+|Percona XtraBackup| has also implemented ``--lock-ddl-per-table``, which blocks DDL statements but allows metadata locks (MDL).
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/advanced/locks.rst]:
```suggestion
|Percona XtraBackup| has also implemented ``--lock-ddl-per-table``, which blocks DDL statements by using metadata locks (MDL).
```

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+The following procedures describe a simplified backup operation:
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/advanced/locks.rst]:
```suggestion
The following procedures describe a simplified backup operation when using ``--lock-ddl-per-table``:
```

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+``--lock-ddl-per-table`` redesign
+----------------------------------
+
+Implemented in |Percona XtraBackup| version 2.4.21, the ``--lock-ddl-per-table`` has been redesigned to minimize inconsistent backups. The following procedure reorders the steps:
+
+* The MDL lock acquired at the beginning of the backup
+
+* Scan the redo logs. An ``MLOG_INDEX_LOAD`` event may be recorded if a ``CREATE INDEX`` statement has occurred before the backup starts. At this time, the backup process is safe and can parse and accept the event.
+
+* After the first scan has started, the follow new redo log events thread is initiated. This thread stops the backup process if an ``MLOG_INDEX_LOAD`` event is found.
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/advanced/locks.rst]:
```suggestion
* After the first scan has completed, the follow redo log thread is initiated. This thread stops the backup process if an ``MLOG_INDEX_LOAD`` event is found.
```

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst (hunk 2 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/locks.rst]: Updated.

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst (hunk 3 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/locks.rst]: updated.

File: storage/innobase/xtrabackup/doc/source/advanced/locks.rst (hunk 1 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/advanced/locks.rst]: updated.

**Review by altmannmarcelo — APPROVED** [review]
LGTM

**Pull request merged**